scripts/bump_aea_version.py

```diff
@@ -18,178 +18,467 @@
+  -h, --help            show this help message and exit
+  --new-version NEW_VERSION
+                        The new AEA version.
+  -p KEY=VALUE [KEY=VALUE ...], --plugin-new-version KEY=VALUE [KEY=VALUE ...]
+                        Set a number of key-value pairs plugin-name=new-
+                        plugin-version
+  --no-fingerprint
+
+Example of usage:
+
+python scripts/bump_aea_version.py --new-version 1.1.0 -p aea-ledger-fetchai=2.0.0 -p aea-ledger-ethereum=3.0.0
+"""
 
 import argparse
+import inspect
+import logging
+import operator
+import os
 import re
 import sys
+from functools import wraps
 from pathlib import Path
+from typing import Any, Callable, Collection, Dict, List, Optional, cast
 
+from git import Repo
+from packaging.specifiers import SpecifierSet
 from packaging.version import Version
 
-from aea.configurations.constants import (
-    DEFAULT_AEA_CONFIG_FILE,
-    DEFAULT_CONNECTION_CONFIG_FILE,
-    DEFAULT_CONTRACT_CONFIG_FILE,
-    DEFAULT_PROTOCOL_CONFIG_FILE,
-    DEFAULT_SKILL_CONFIG_FILE,
-)
 from aea.helpers.base import compute_specifier_from_version
 from scripts.generate_ipfs_hashes import update_hashes
 
 
-VERSION_NUMBER_PART_REGEX = r"(0|[1-9]\d*)"
-VERSION_REGEX = fr"(any|latest|({VERSION_NUMBER_PART_REGEX})\.({VERSION_NUMBER_PART_REGEX})\.({VERSION_NUMBER_PART_REGEX})(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)"
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s][%(name)s][%(levelname)s] %(message)s"
+)
+
+# if the key is a file, just process it
+# if the key is a directory, process all files below it
+PatternByPath = Dict[Path, str]
 
-PACKAGES_DIR = Path("packages")
-TESTS_DIR = Path("tests")
 AEA_DIR = Path("aea")
-CONFIGURATION_FILENAME_REGEX = re.compile(
-    "|".join(
-        [
-            DEFAULT_AEA_CONFIG_FILE,
-            DEFAULT_SKILL_CONFIG_FILE,
-            DEFAULT_CONNECTION_CONFIG_FILE,
-            DEFAULT_CONTRACT_CONFIG_FILE,
-            DEFAULT_PROTOCOL_CONFIG_FILE,
-        ]
-    )
+CUR_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))  # type: ignore
+ROOT_DIR = Path(os.path.join(CUR_PATH, ".."))
+
+PLUGINS_DIR = Path("plugins")
+ALL_PLUGINS = tuple(PLUGINS_DIR.iterdir())
+
+"""
+This pattern captures a specifier set in the dependencies section
+of an AEA package configuration file, e.g.:
+
+dependencies:
+    ...
+    aea-ledger-fetchai:
+        version: >=1.0.0,<2.0.0
+"""
+YAML_DEPENDENCY_SPECIFIER_SET_PATTERN = (
+    "(?<={package_name}:\n    version: )({specifier_set})"
 )
 
-IGNORE_DIRS = [Path(".git")]
+"""
+This pattern captures a specifier set for PyPI dependencies
+in JSON format.
 
+e.g.:
+"aea-ledger-fetchai": {"version": ">=2.0.0, <3.0.0"}
+"""
+JSON_DEPENDENCY_SPECIFIER_SET_PATTERN = (
+    '(?<="{package_name}": ."version": ")({specifier_set})(?=".)'
+)
 
-def update_version_for_files(current_version: str, new_version: str) -> None:
```

Comment:
the additions below are mostly a refactoring of the old implementation. Unfortunately git changes are not very helpful